### PR TITLE
DAOS-12121 test: Bullseye backports from master branch to rel_2.2

### PR DESCRIPTION
**Pull request by dinghwah**
Skip-fnbullseye: false
Skip-bullseye: false
Skip-python-bandit: true
Skip-build-leap15-rpm: true
Skip-build-ubuntu20-rpm: true
Allow-unstable-test: true
Doc-only: false
Test-tag: pr daily_regression
Required-githooks: true
Signed-off-by: Ding Ho ding-hwa.ho@intel.com

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -265,7 +265,7 @@ pipeline {
                     env.COMMIT_MESSAGE.split('\n').each { line ->
                         String key, value
                         try {
-                            (key, value) = line.split(':')
+                            (key, value) = line.split(':', 2)
```
Comment (brianjmurrell) [inline Jenkinsfile:268]: This is obsolete now.  You need to replace the entire block: https://github.com/daos-stack/daos/blob/1a149dd13cf39c0b861686c6529d6bcc4c9c6902/Jenkinsfile#L260-L279 with simply:

```
                pragmasToEnv()
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
The CI parts LGTM.  I cannot comment on the remaining python changes.

This cannot land however, until https://github.com/daos-stack/spdk/pull/56 lands and that PR's latest test run hit a [lot of failures](https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/ci-daos-stack-spdk-PR-56-master/4/tests/) on daos master that will need to be understood (i.e. are they known test failures on daos master?) before it can be landed.

**Comment by dinghwah** [comment]
Python scripts change and test-logs are consistent with Master branch.

**Review by daltonbohning — APPROVED** [review]

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
-1 until https://github.com/daos-stack/spdk/pull/56 lands which has been hitting various issues in it's testing.

**Review by shimizukko — APPROVED** [review]

**Comment by brianjmurrell** [comment]
> brianjmurrell requested a review from shimizukko now

Apologies.  Requesting another review form @shimizukko was a mis-click on my part.  Slow GUI update meant I clicked somewhere I didn't think I was clicking.  Could you please put your +1 back @shimizukko.  Thanks!

**Review by brianjmurrell — APPROVED** [review]
https://github.com/daos-stack/spdk/pull/56 has landed.  It will take some time for it to be built and uploaded to Artifactory, but in a while you should be able to remove the `PR-repos:` from this PR and get the new spdk @dinghwah.

**Review by shimizukko — APPROVED** [review]

**Comment by brianjmurrell** [comment]
@dinghwah Your most recent commit on this PR still has a `PR-repos` commit pragma in it.  @daos-stack/daos-gatekeeper should not land PRs with such a commit pragma in their last commit and they don't know if this will cause breakage when they remove it on landing.

Please remove the pragma in a new commit so that this runs without the need for an spdk PR.

**Comment by dinghwah** [comment]
2 failures are due to:
(1)DAOS-12292: HDF5 test timeout
(2)DAOS-12199: ior: dfs_open() of testfile.00000001 Failed

**Pull request merged**